librz/analysis/arch_profile.c

```diff
@@ -0,0 +1,133 @@
+// SPDX-FileCopyrightText: 2021 RizinOrg <info@rizin.re>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_types.h>
+#include <rz_util.h>
+#include <rz_arch.h>
+#include <stdio.h>
+#include <string.h>
+
+RZ_API void rz_arch_profile_free(RzArchProfile *p) {
+	if (!p) {
+		return;
+	}
+	ht_up_free(p->registers_mmio);
+	ht_up_free(p->registers_extended);
+}
+
+RZ_API RZ_OWN RzArchProfile *rz_arch_profile_new() {
```

Comment:
Don't forget Doxygen documentation for newly added RZ_API

Reply:
Sure, I'll add them for all of the new ones.